### ISSUE TITLE
build: add odd validation to GH Actions 

### ISSUE THREAD

**Issue by rettinghaus** [music-encoding/music-encoding#790]
It would be nice if we could had a validation of the schema for every PR.

**Comment by kepper** [comment]
I totally agree, and @o-sapov has prepared something a while ago – I just need to go back to it and have a closer look to make it happen. Don't know how to integrate that with GH actions, though…

**Comment by kepper** [comment]
During ODD Thursday, we agreed to first improve Schematron validation of the sources (with pecial focus on docs), and then as a second step include that as a Github Action.

**Comment by musicEnfanthen** [comment]
In fact, there was something like schema validation with every PR via TravisCI: cf. https://github.com/music-encoding/music-encoding/blob/develop/CONTRIBUTING.md#travisci

**Comment by musicEnfanthen** [comment]
The corresponding files were removed in https://github.com/music-encoding/music-encoding/commit/ea749ca0f2351f0a7875c572635993a3baed0e0e and https://github.com/music-encoding/music-encoding/commit/fe34a528d9feb0452f36748401b06277bb50ecba .

Maybe we could reuse/transform these scripts to GH actions?

**Comment by kepper** [comment]
There is a first draft of the Schematron validation of the ODD sources. It has already surfaced a couple of inconsistencies that should be addressed before merging this as a PR. We also need to see how it can be integrated into the GH Actions.

**Comment by musicEnfanthen** [comment]
See also #264 and #806

**Comment by musicEnfanthen** [comment]
Integration with GH actions not solved yet

**Comment by musicEnfanthen** [comment]
I suggest to move this to 5.x

**Comment by bwbohl** [comment]
moved it

**Comment by musicEnfanthen** [comment]
Eventually I have figured out how to integrate schematron validation into the ant `build.xml` script. It uses a jar file for [`schxslt`](https://github.com/schxslt/schxslt/tree/master/ant) and runs `mei-source.sch` on the `canonicalized-source.xml`.

```
    <target name="validate-source" description="Validates the canonicalized source.">
        <taskdef name="schematron" classname="name.dmaus.schxslt.ant.Task" classpath="${dir.lib.schematron}/${schematron.jar.file}"/>
        <schematron schema="${dir.validation}/mei-source.sch" file="${canonicalized.source.path}" report="${dir.build}/validation-report.xml"/>
    </target>
```

I'm just not sure if this ant version of schxslt is still supported or maintained. But maybe @dmj can tell more about this?

**Comment by dmj** [comment]
SchXslt is still maintained and so is the Ant task.

**Comment by musicEnfanthen** [comment]
Thank you @dmj for confirmation 🙏  I was just wondering since the releases section of the ant schxslt says that the repo was archived in 2020? (https://github.com/schxslt/schxslt-ant/releases) And while the main repo is on v1.9.5, the most recent version of those ant jars is v1.4? Is there another more recent jar file to be preferred?

**Comment by rettinghaus** [comment]
@musicEnfanthen Would you be so kind to open a PR for local testing?

**Comment by dmj** [comment]
> Thank you @dmj for confirmation 🙏 I was just wondering since the releases section of the ant schxslt says that the repo was archived in 2020? (https://github.com/schxslt/schxslt-ant/releases) And while the main repo is on v1.9.5, the most recent version of those ant jars is v1.4? Is there another more recent jar file to be preferred?

Yikes! Now I understand your question! The schxslt-ant repository is indeed archived. Its source was moved to the main SchXslt repository. 

The releases package of SchXslt https://github.com/schxslt/schxslt/releases provides a .jar with the Ant step, i.e. ant-schxslt-1.9.5.jar is the most recent one.

**Comment by musicEnfanthen** [comment]
@dmj Ah, I see. That's great and good to know. Will give it a try, thank you!

EDIT: It seems that my confusion came from the README here ([https://github.com/schxslt/schxslt/blob/master/ant/README.md](https://github.com/schxslt/schxslt/blob/master/ant/README.md)) pointing to the archived  ant-schxslt repo.

**Comment by musicEnfanthen** [comment]
@rettinghaus Yes, will be happy to do by the end of the week.

**Comment by musicEnfanthen** [comment]
@bwbohl can you have a look at #1420, please . Would need that parametrization first.

**Comment by bwbohl** [comment]
done, still waiting for a second approval of #1420

**Comment by musicEnfanthen** [comment]
@rettinghaus @bwbohl @kepper Draft to play with prepared in #1428

**Comment by bwbohl** [comment]
@bwbohl pinging myself
